src/extends/room/movement.js

```diff
@@ -0,0 +1,115 @@
+'use strict'
+
+Room.getCostmatrix = function (roomname, opts = {}) {
+  Logger.log(`Retrieving CostMatrix for ${roomname}`, LOG_INFO)
+
+  const cm = Room.getStructuresCostmatrix(roomname, opts)
+
+  // Add in creeps
+  if (!opts.ignoreCreeps && Game.rooms[roomname]) {
+    const room = Game.rooms[roomname]
+    const creeps = room.find(FIND_CREEPS)
+    for (let creep of creeps) {
+      cm.set(creep.pos.x, creep.pos.y, 255)
+    }
+  }
+
+  // Block all of these positions as unwalkable
+  if (opts.avoid) {
```

Comment:
Should perhaps this also be moved below the application of avoidRange, since avoidRange can set lower values?

Reply:
I did it about two seconds before you made this comment here :-)

Reply:
Good, just figured I'd put my comment into github as well to be thorough for posterity. :)